course3/homework-vector/vector.cc

```diff
@@ -1,9 +1,53 @@
 #include <iostream>
 #include <cassert>
+#include <cmath>
 
 class Vector
 {
-    // TODO
+    private:
+        int private_x;
+        int private_y;
+
+    public:
+        // required constructors
+        Vector(int x, int y): private_x(x), private_y(y) {}
+        Vector() = default;
```

Comment:
The value of a default constructed int is 0.

On Thu, Apr 14, 2016 at 1:42 PM maciekgajewski <notifications@github.com>
wrote:

> In course3/homework-vector/vector.cc
> <https://github.com/maciekgajewski/c-excercises/pull/10#discussion_r59702654>
> :
>
> >
> >  class Vector
> >  {
> > -    // TODO
> > +    private:
> > +        int private_x;
> > +        int private_y;
> > +
> > +    public:
> > +        // required constructors
> > +        Vector(int x, int y): private_x(x), private_y(y) {}
> > +        Vector() = default;
>
> How will this initialize the members? what the values of private_x and
> private_y will be in default-constructed object?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/maciekgajewski/c-excercises/pull/10/files/0c2ed6bfc9ba8aec2d7fec3df378308e23f7d993#r59702654>
>